presto-main/src/main/java/com/facebook/presto/sql/gen/PageProcessorCompiler.java

```diff
@@ -95,17 +95,28 @@ public PageProcessorCompiler(Metadata metadata)
     @Override
     public void generateMethods(ClassDefinition classDefinition, CallSiteBinder callSiteBinder, RowExpression filter, List<RowExpression> projections)
     {
-        ImmutableList.Builder<MethodDefinition> projectionMethods = ImmutableList.builder();
+        ImmutableList.Builder<MethodDefinition> projectMethods = ImmutableList.builder();
+        ImmutableList.Builder<MethodDefinition> projectColumnarMethods = ImmutableList.builder();
+        ImmutableList.Builder<MethodDefinition> projectDictionaryMethods = ImmutableList.builder();
         for (int i = 0; i < projections.size(); i++) {
-            projectionMethods.add(generateProjectMethod(classDefinition, callSiteBinder, "project_" + i, projections.get(i)));
+            MethodDefinition project = generateProjectMethod(classDefinition, callSiteBinder, "project_" + i, projections.get(i));
+            MethodDefinition projectColumnar = generateProjectColumnarMethod(classDefinition, callSiteBinder, "projectColumnar_" + i, projections.get(i), project);
+            MethodDefinition projectDictionary = generateProjectDictionaryMethod(classDefinition, "projectDictionary_" + i, projections.get(i), project, projectColumnar);
+
+            projectMethods.add(project);
```

Comment:
I reuse the project and projectColumnar, so I did the same for projectDictionary to keep it consistent.